src/Microsoft.DotNet.Internal.Tools.SynchronizePackageProps/Program.cs

```diff
@@ -0,0 +1,397 @@
+                IReadOnlyList<RepositoryContent> contents;
+                string owner = repoMatch.Groups[1].Value;
+                string repository = repoMatch.Groups[2].Value;
+                try
+                {
+                    contents = await client.Repository.Content.GetAllContentsByRef(owner,
+                        repository,
+                        "eng/Packages.props",
+                        dep.CommitHash);
+                }
+                catch (NotFoundException)
+                {
+                    try
+                    {
+                        contents = await client.Repository.Content.GetAllContentsByRef(owner,
+                            repository,
+                            "Packages.props",
+                            dep.CommitHash);
+                    }
+                    catch (NotFoundException)
+                    {
+                        WriteWarning($"No Packages.props found for '{repo}' at '{dep.CommitHash}'");
+                        continue;
+                    }
+                }
+
+                Dictionary<string, string> result;
+                if (string.IsNullOrEmpty(contents[0].Content))
+                {
+                    if (contents[0].Size > 100000)
+                    {
+                        WriteWarning($"Packages.props too large ({contents[0].Size}) in {repo}, skipping...");
+                        continue;
+                    }
+
+                    result = ParsePackageVersions(XDocument.Load(contents[0].DownloadUrl), $"{owner}/{repository}");
+                }
+                else
+                {
+                    result = ParsePackageVersions(XDocument.Parse(contents[0].Content), $"{owner}/{repository}");
+                }
+
+                remotePackages.Add(repo, result);
+            }
+
+            return remotePackages;
+        }
+
+        private static Dictionary<string, string> ParsePackageVersions(XDocument packagesDocument, string repoName)
+        {
+            var props = new Dictionary<string, string>();
+            IEnumerable<XElement> propGroups = packagesDocument?.Element("Project")?.Elements("PropertyGroup");
+            if (propGroups != null)
+            {
+                foreach (XElement propGroup in propGroups)
+                {
+                    if (propGroup.Attribute("Condition") != null)
+                    {
+                        WriteWarning($"{repoName} Packages.prop has Conditional PropertyGroup, which is not supported");
+                        continue;
+                    }
+
+                    foreach (XElement prop in propGroup.Elements())
+                    {
+                        if (prop.Attribute("Condition") != null)
+                        {
+                            WriteWarning(
+                                $"{repoName} Packages.prop property {prop.Name.LocalName} has Conditional PropertyGroup, which is not supported");
```

Comment:
>PropertyGroup [](start = 106, length = 13)

NIT: this is an error for a Conditional property, not a conditional property group.